README.md

```diff
@@ -1,71 +1,156 @@
+
+## Rollout Policies
+
+### Example Policies
+
+KCD comes with some predefined, example policies for you to play with. One example is the following yaml manifest of a policy:
+
+```yaml
+apiVersion: delivery.knative.dev/v1alpha1
+kind: Policy
+metadata:
+  namespace: default
+  name: advanced-policy
+spec:
+  mode: time
+  defaultThreshold: 60
+  stages:
+  - percent: 25
+    threshold: 30
+  - percent: 50
+    threshold: 30
+  - percent: 75
+```
+
+The `mode` field specifies how the threshold values should be interpreted. A time-based policy implies that the threshold values should be interpreted with units of seconds. A threshold value is used to determine when to move the rollout into the next stage. The first rollout stage in the example above has a rollout threshold of 30 seconds, meaning that 30 seconds after the rollout enters this stage, it will progress to the next stage.
+
+The `defaultThreshold` field specifies what the threshold value for a stage should be if that stage does not specify an optional threshold value on its own. For example, the third stage in the example above (with 75%) does not have an optional threshold value, so its threshold is 60 seconds.
+
+Each `percent` in the list of stages specify the rollout percentage for that stage, i.e. the percentage of traffic that the newest Revision should be receive at this stage.
+
+### Customize Policies
+
+Although you can choose to use one of the example policies directly, you can also write and apply rollout policies on your own. To do so, you need a yaml file that describes your policy in a format similar to the one shown above. A policy can only affect the rollout behavior of things that have the same namespace, meaning that even if a Service in a different namespace annotates a certain policy, that Service's traffic behavior will not be affected.
+
+When applying policies, the following criteria must be met (otherwise your policy will be rejected and will have no effect):
+
+1. The value of the `mode` field must be valid (currently only "time" is accepted).
+2. The value of `defaultThreshold` must be a positive integer, and `defaultThreshold` cannot be omitted.
+3. There must be at least one rollout stage.
+4. The values of `percent` fields must be in non-decreasing order.
+5. The values of `percent` fields must be in the range [1, 99]. You do not need to specify 0 as the first stage, and you MUST not specify 100 as the final stage.
+6. Any optional `threshold` values, if specified, must be positive integers.
+
+## Example Workflow
+
+### Installing KCD
+
+Before installing and using KCD, you need to make sure that you have `go`, `kubectl`, and `ko` installed on your machine. Your cluster must also have Knative.
```

Comment:
nit: the cluster must have knative serving, not just knative.

Reply:
fixed